src/rpc/methods/eth.rs

```diff
@@ -1259,6 +1259,47 @@ impl RpcMethod<2> for EthGetBlockByNumber {
     }
 }
 
+pub enum EthGetBlockReceipts {}
+impl RpcMethod<1> for EthGetBlockReceipts {
+    const NAME: &'static str = "Filecoin.EthGetBlockReceipts";
+    const NAME_ALIAS: Option<&'static str> = Some("eth_getBlockReceipts");
+    const PARAM_NAMES: [&'static str; 1] = ["block_hash"];
+    const API_PATHS: ApiPaths = ApiPaths::V1;
+    const PERMISSION: Permission = Permission::Read;
+
+    type Params = (EthHash,);
+    type Ok = Vec<EthTxReceipt>;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (block_hash,): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        let ts = get_tipset_from_hash(ctx.chain_store(), &block_hash)?;
+
+        let (_, msgs_and_receipts) = execute_tipset(&ctx, &Arc::new(ts.clone())).await?;
+
+        let mut receipts = Vec::with_capacity(msgs_and_receipts.len());
+
+        for (i, (msg, receipt)) in msgs_and_receipts.into_iter().enumerate() {
+            let return_data = receipt.return_data().clone();
```

Comment:
You removed one clone but added another one in the process. You can avoid cloning entirely by computing `return_dec` directly.

Reply:
Oh right, I needed to do it before MessageLookup